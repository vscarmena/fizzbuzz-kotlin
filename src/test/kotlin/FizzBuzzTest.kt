import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance
import kotlin.test.assertEquals

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class FizzBuzzTest {

    @Test
    fun `evaluate numbers to apply conversion`() {
        Approvals.verifyAll(
            intArrayOf(1, 2, 3, 4, 5, 6, 10, 15, 20, 30, 50, 98, 99, 100).toTypedArray(),
            this::getGeneratedValue
        )
    }

    private fun getGeneratedValue(number: Int): String {
        return FizzBuzz().generateValues()[number - 1]
    }

    @Test
    fun `generate values to print`() {
        assertEquals(100, FizzBuzz().generateValues().size)
    }

}