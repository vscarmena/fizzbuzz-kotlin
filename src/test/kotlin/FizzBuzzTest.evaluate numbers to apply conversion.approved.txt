1
2
Fizz
4
Buzz
Fizz
Buzz
FizzBuzz
Buzz
FizzBuzz
Buzz
98
Fizz
Buzz
